### PR TITLE
add fuse admm/matmul back to linear pass

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58 add fuse admm/matmul back to linear pass**
* #57 [codemod] deprecated macros

### CODE REVIEW

File: torch_tvm/fuse_linear.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    std::string addmm_pattern = R"IR(
+    graph(%input, %weight, %bias, %4):
+        %weight_t = aten::t(%weight)
+        %res = aten::addmm(%bias, %input, %weight_t, %4, %4)
+        return (%res))IR";
+    std::string matmul_add_pattern = R"IR(
+    graph(%input, %weight, %bias, %4):
+        %weight_t = aten::t(%weight)
+        %output = aten::matmul(%input, %weight_t)
+        %res = aten::add_(%output, %bias, %4)
```
Comment (bwasti) [inline torch_tvm/fuse_linear.cpp]: should we also match on non-inplace add?

File: test/test_operators.py [hunk 2]
```diff
@@ -206,9 +206,15 @@ def test_linear(self, shape, out_features):
         def linear(input, weight, bias):
             return F.linear(input, weight, bias) + 2.0
 
+        def linear_no_bias(input, weight):
```
Comment (bwasti) [inline test/test_operators.py]: maybe we should test the pattern matching in python?

**Review by bwasti — APPROVED** [review]
nice! :)

File: test/test_operators.py (hunk 2 above)
Comment (wanchaol) [reply test/test_operators.py]: Yeah sure we will need to pybind the pass to python, I will add a test

File: torch_tvm/fuse_linear.cpp (hunk 1 above)
Comment (wanchaol) [reply torch_tvm/fuse_linear.cpp]: We can but I don't think we need, people using `nn.Linear` will always get this inplace `add` through tracing/scripting, outplace `add` will be matched if people manually write `matmul + outplace add`.

File: test/test_operators.py (hunk 2 above)
Comment (bwasti) [reply test/test_operators.py]: can we avoid pybind by manually constructing a torchscript model containing t + matmul + _add and just ensuring "linear" exists in the graph_for call?

**Pull request closed without merging**